### PR TITLE
Add maven plugin for documenting conf properties

### PR DESCRIPTION
**Pull request by ericbottard**
Requires https://github.com/spring-cloud/spring-cloud-dataflow/pull/715

### CODE REVIEW

**Comment by ericbottard** [comment]
This can't be applied to the RC1 release, as it depends on a dataflow change that deliberately chose not to include in RC1

**Comment by ericbottard** [comment]
This can be ran locally though, to find and catch documentation issues that will be fixed for RC1

**Comment by ericbottard** [comment]
Rebased, ran against latest starters

**Comment by ilayaperumalg** [comment]
@ericbottard I get compilation error here:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project spring-cloud-stream-app-documentation-maven-plugin: Compilation failure
[ERROR] /Users/igopinatha/workspace/git/spring-cloud/spring-cloud-stream-app-starters/spring-cloud-stream-app-documentation-maven-plugin/src/main/java/org/springframework/cloud/stream/app/documentation/plugin/ConfigurationMetadataDocumentationMojo.java:[81,106] incompatible types: org.springframework.cloud.stream.app.documentation.plugin.ConfigurationMetadataDocumentationMojo.ScatteredArchive cannot be converted to org.springframework.core.io.Resource
[ERROR] -> [Help 1]
[ERROR]
```

File: spring-cloud-stream-app-documentation-maven-plugin/src/main/java/org/springframework/cloud/stream/app/documentation/plugin/ConfigurationMetadataDocumentationMojo.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
+package org.springframework.cloud.stream.app.documentation.plugin;
```
Comment (ilayaperumalg) [inline spring-cloud-stream-app-documentation-maven-plugin/src/main/java/org/springframework/cloud/stream/app/documentation/plugin/ConfigurationMetadataDocumentationMojo.java:16]: Copyright header.

**Comment by ericbottard** [comment]
@markpollack I think I need to clarify that this cannot be merged and expected to be part of a release of starters until we also cut any kind of release for dataflow that encompasses the method signature changes :(

**Comment by ilayaperumalg** [comment]
@ericbottard Since the plugin generates doc only if the apps have `README` file, we might be missing doc for some apps. For example, the doc for `aggregate-counter` is missing inside `metrics`.

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as fc4d2f029789cf5b42e74db77d4ada3620a70f15 along with the doc update for `triggertask` adc5133ebbffd10939d43bc0280f5a3ec667099e

**Pull request closed without merging**